app/models/user.rb

```diff
@@ -76,19 +76,24 @@ def roles(_course)
   end
 
   def contribution_url
-    language = ENV['wiki_language']
-    "https://#{language}.wikipedia.org/wiki/Special:Contributions/#{username}"
+    "#{home_wiki.base_url}/wiki/Special:Contributions/#{wiki_id}"
   end
 
+  # Provides a link to the list of all of a user's subpages, ie, sandboxes.
   def sandbox_url
-    language = ENV['wiki_language']
-    "https://#{language}.wikipedia.org/wiki/Special:PrefixIndex/User:#{username}"
+    "#{home_wiki.base_url}/wiki/Special:PrefixIndex/User:#{wiki_id}"
   end
 
   def talk_page
     "User_talk:#{username}"
   end
 
+  def home_wiki
```

Comment:
Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)